enzyme/Enzyme/GradientUtils.h

```diff
@@ -192,6 +192,8 @@ class GradientUtils : public CacheUtility {
   getInvertedBundles(CallInst *orig, ArrayRef<ValueType> types,
                      IRBuilder<> &Builder2, bool lookup,
                      const ValueToValueMapTy &available = ValueToValueMapTy()) {
+    assert(!(lookup && mode == DerivativeMode::ForwardMode));
```

Comment:
@wsmoses this assertion should hold right? Because it doesn't hold for:

  Enzyme :: Enzyme/ForwardMode/badcall.ll
  Enzyme :: Enzyme/ForwardMode/badcall2.ll
  Enzyme :: Enzyme/ForwardMode/badcall3.ll
  Enzyme :: Enzyme/ForwardMode/badcall4.ll
  Enzyme :: Enzyme/ForwardMode/badcallsq.ll
  Enzyme :: Enzyme/ForwardMode/badcallused.ll
  Enzyme :: Enzyme/ForwardMode/badcallused2.ll
  Enzyme :: Enzyme/ForwardMode/call.ll
  Enzyme :: Enzyme/ForwardMode/callmincacheunwrap.ll
  Enzyme :: Enzyme/ForwardMode/callmod.ll
  Enzyme :: Enzyme/ForwardMode/custom0.ll
  Enzyme :: Enzyme/ForwardMode/custom1.ll
  Enzyme :: Enzyme/ForwardMode/custom2.ll
  Enzyme :: Enzyme/ForwardMode/globalfn.ll
  Enzyme :: Enzyme/ForwardMode/ptr-ret.ll
  Enzyme :: Enzyme/ForwardMode/relu.ll
  Enzyme :: Enzyme/ForwardMode/square2.ll

Reply:
It should...I would see how are you calling it to trigger it?

Reply:
You pass `/*lookup*/ true` when handling forward mode calls.

Reply:
https://github.com/EnzymeAD/Enzyme/pull/520